### PR TITLE
Gracefully close the app on exit

### PR DESCRIPTION
**Pull request by kentaromiura**
implement http://switchbrew.org/index.php?title=SVC#svcExitProcess as application exit to avoid NotImplemented Exception

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -50,6 +50,10 @@ static void Main(string[] args)
 
             using (GLScreen Screen = new GLScreen(Ns, Renderer))
             {
+                Ns.Finish += (Sender, Args) => {
```
Comment (AcK77) [inline Program.cs]: Please put the { on a new line.

File: Ryujinx/OsHle/Horizon.cs [hunk 2]
```diff
@@ -136,6 +137,18 @@ public void FinalizeAllProcesses()
             }
         }
 
+        internal bool ExitProcess(int ProcessId) {
```
Comment (AcK77) [inline Ryujinx/OsHle/Horizon.cs]: Please put the { on a new line.

File: Ryujinx/OsHle/Horizon.cs [hunk 3]
```diff
@@ -136,6 +137,18 @@ public void FinalizeAllProcesses()
...
+            Process process;
+            var Success = Processes.TryRemove(ProcessId, out process);
+            if (Success) {
```
Comment (AcK77) [inline Ryujinx/OsHle/Horizon.cs]: Delete all "if" {} or put the { on a new line.

File: Ryujinx/OsHle/Horizon.cs [hunk 4]
```diff
@@ -136,6 +137,18 @@ public void FinalizeAllProcesses()
...
+                process.StopAllThreads();
+            }
+
+            if (Processes.Count == 0) {
```
Comment (AcK77) [inline Ryujinx/OsHle/Horizon.cs]: Delete all "if" {} or put the { on a new line.

**Comment by gdkchan** [comment]
Thanks! In the future please conform with the style used on the rest of the code (like `{` on a new line, space between variables/functions and so on).

**Pull request merged**